modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/controller/CheckerConfigurationController.scala

```diff
@@ -153,93 +158,95 @@ class CheckerConfigurationController {
-      }
     } else {
       throw new ForbiddenException()
     }
   }
 
   /**
     * Upload a the main file for a task configuration
-    * @param cid Course id
-    * @param tid Task id
+    *
+    * @param cid  Course id
+    * @param tid  Task id
     * @param ccid Checker configuration id
     * @param file Multipart file
-    * @param req Http request
-    * @param res Http response
+    * @param req  Http request
+    * @param res  Http response
     */
   @PutMapping(value = Array("/{cid}/tasks/{tid}/checker-configurations/{ccid}/main-file"))
   def updateMainFile(@PathVariable cid: Int, @PathVariable tid: Int, @PathVariable ccid: Int,
                      @RequestParam file: MultipartFile,
                      req: HttpServletRequest, res: HttpServletResponse): Unit =
-    uploadFile(storageService.storeMainFile,
+    uploadFile("main-file",
       (cc) => {
-        notifyCheckerMainFileUpload(cid, taskService.getOne(tid).get, cc, storageService.pathToMainFile(ccid).get)
+        // ?? //notifyCheckerMainFileUpload(cid, taskService.getOne(tid).get, cc, storageService.pathToMainFile(ccid).get)
         this.ccs.setMainFileUploadedState(cid, tid, ccid, state = true)
       })(cid, tid, ccid, file, req, res)
 
   /**
     * Downloads the main file for a task configuration
-    * @param cid Course id
-    * @param tid Task id
+    *
+    * @param cid  Course id
+    * @param tid  Task id
     * @param ccid Checker configuration id
-    * @param req Http request
-    * @param res Http response
+    * @param req  Http request
+    * @param res  Http response
     */
   @GetMapping(value = Array("/{cid}/tasks/{tid}/checker-configurations/{ccid}/main-file"))
   def getMainFile(@PathVariable cid: Int, @PathVariable tid: Int, @PathVariable ccid: Int,
-                     req: HttpServletRequest, res: HttpServletResponse): Unit =
-    getFile(storageService.pathToMainFile)(cid, tid, ccid, req, res)
+                  req: HttpServletRequest, res: HttpServletResponse): Unit =
+    getFile(storageService.pathToMainFile)("main-file", cid, tid, ccid, req, res)
 
-    /**
+  /**
     * Upload a the secondary file for a task configuration
-    * @param cid Course id
-    * @param tid Task id
+    *
+    * @param cid  Course id
+    * @param tid  Task id
     * @param ccid Checker configuration id
     * @param file Multipart file
-    * @param req Http request
-    * @param res Http response
+    * @param req  Http request
+    * @param res  Http response
     */
   @PutMapping(value = Array("/{cid}/tasks/{tid}/checker-configurations/{ccid}/secondary-file"))
   def uploadSecondaryFile(@PathVariable cid: Int, @PathVariable tid: Int, @PathVariable ccid: Int,
-                         @RequestParam file: MultipartFile,
+                          @RequestParam file: MultipartFile,
                           req: HttpServletRequest, res: HttpServletResponse): Unit =
-    uploadFile(storageService.storeSecondaryFile,
+    uploadFile("secondary-file",
```

Comment:
I would store the name of the `secondary-file` in as a static member of a global object